### PR TITLE
add fast-unstsake pallet to all runtimes

### PR DESCRIPTION
**Pull request by kianenigma**
https://github.com/paritytech/substrate/pull/12129
Substrate dependencies updates to https://github.com/paritytech/substrate/pull/12344

- Adds the fast-unstake pallet to all 3 runtimes. The pallet will in itself be in an idle state where no unstake requests can be processed. 

- A separate governance proposal, which can be controlled by root or 1/2 of council (see `ControlOrigin`) can set the number of eras that are checked per block, which will set this pallet into motion. Until then, stakers can register, but nothing will be processed. 

- `FastUnstake` calls are part of the staking proxy, since the normal proxy can already fully unbond a stash.

### CODE REVIEW

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime polkadot-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime kusama-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime westend-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime westend-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime kusama-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime polkadot-dev pallet_fast_unstake
/cmd queue -c bench-bot $ runtime kusama-dev pallet_fast_unstake
/cmd queue -c bench-bot $ runtime westend-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime kusama-dev pallet_fast_unstake

File: runtime/polkadot/src/lib.rs [hunk 1]
```diff
@@ -566,6 +566,17 @@ impl pallet_staking::Config for Runtime {
 	type WeightInfo = weights::pallet_staking::WeightInfo<Runtime>;
 }
 
+impl pallet_fast_unstake::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	// slash by 1 DOTs per era, if they are proven to be wrong.
```
Comment (BustaNit) [inline runtime/polkadot/src/lib.rs:571]:
```suggestion
	// slash by 1 DOT per era, if they are proven to be wrong.
```

File: runtime/westend/src/lib.rs [hunk 2]
```diff
@@ -517,6 +517,14 @@ impl pallet_staking::Config for Runtime {
 	type WeightInfo = weights::pallet_staking::WeightInfo<Runtime>;
 }
 
+impl pallet_fast_unstake::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	// slash by 1 WNDs per era, if they are proven to be wrong.
```
Comment (BustaNit) [inline runtime/westend/src/lib.rs:522]:
```suggestion
	// slash by 1 WND per era, if they are proven to be wrong.
```

**Review by BustaNit — APPROVED** [review]
Busted 2 nits, otherwise LGTM

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime kusama-dev pallet_fast_unstake

**Review by ggwpez — APPROVED** [review]

**Review by ordian — APPROVED** [review]

File: runtime/kusama/src/weights/pallet_fast_unstake.rs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	// Storage: FastUnstake Head (r:1 w:1)
+	// Storage: Staking CurrentEra (r:1 w:0)
+	// Storage: Staking SlashingSpans (r:1 w:0)
+	// Storage: Staking Bonded (r:2 w:1)
+	// Storage: Staking Ledger (r:2 w:2)
+	// Storage: Staking Validators (r:1 w:0)
+	// Storage: Staking Nominators (r:1 w:0)
+	// Storage: System Account (r:3 w:2)
+	// Storage: Balances Locks (r:2 w:2)
+	// Storage: NominationPools MinJoinBond (r:1 w:0)
```
Comment (kianenigma) [inline runtime/kusama/src/weights/pallet_fast_unstake.rs:60]: we should see all of this gone now.

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime kusama-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime kusama-dev pallet_fast_unstake

**Review by Ank4n — APPROVED** [review]

**Review by rossbulat — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime westend-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
/cmd queue -c bench-bot $ runtime westend-dev pallet_fast_unstake

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**